common/scala/src/main/scala/org/apache/openwhisk/core/database/CouchDbRestClient.scala

```diff
@@ -42,9 +42,11 @@ import scala.concurrent.{ExecutionContext, Future}
 class CouchDbRestClient(protocol: String, host: String, port: Int, username: String, password: String, db: String)(
   implicit system: ActorSystem,
   logging: Logging)
-    extends PoolingRestClient(protocol, host, port, 16 * 1024) {
+    extends PoolingRestClient(protocol, host, port, 16 * 1024)(
+      system,
+      system.dispatchers.lookup("dispatchers.couch-dispatcher")) {
```

Comment:
I couldn't understand why we need to pass this same execution context.

Reply:
The future `onComplete` requires an execution context, but because `onComplete` is defined during class initialization it doesn't have a proper execution context in the case of `CouchDbRestClient` because the child class is initialized later than the parent.

I kept the protected property in `CouchDbRestClient` because its children depend on it, there is at least one implementation inside OW.